kpi/serializers/v2/service_usage.py

```diff
@@ -62,64 +80,180 @@ def get_storage_bytes(self, asset):
         self._total_submission_count_current_month = 0
+        self._total_submission_count_current_year = 0
+        self._current_month_start = None
+        self._current_year_start = None
         self._get_per_asset_usage(instance)
 
-    def get_total_nlp_asr_seconds(self, user):
-        return self._total_nlp_asr_seconds
+    def get_total_nlp_asr_seconds_all_time(self, user):
+        return self._total_nlp_asr_seconds_all_time
+
+    def get_total_nlp_asr_seconds_current_month(self, user):
+        return self._total_nlp_asr_seconds_current_month
+
+    def get_total_nlp_asr_seconds_current_year(self, user):
+        return self._total_nlp_asr_seconds_current_year
+
+    def get_total_nlp_mt_characters_all_time(self, user):
+        return self._total_nlp_mt_characters_all_time
+
+    def get_total_nlp_mt_characters_current_month(self, user):
+        return self._total_nlp_mt_characters_current_month
 
-    def get_total_nlp_mt_characters(self, user):
-        return self._total_nlp_mt_characters
+    def get_total_nlp_mt_characters_current_year(self, user):
+        return self._total_nlp_mt_characters_current_year
 
     def get_total_submission_count_all_time(self, user):
         return self._total_submission_count_all_time
 
     def get_total_submission_count_current_month(self, user):
         return self._total_submission_count_current_month
 
+    def get_total_submission_count_current_year(self, user):
+        return self._total_submission_count_current_year
+
     def get_total_storage_bytes(self, user):
         return self._total_storage_bytes
 
+    def get_current_month_start(self, user):
+        return self._current_month_start
+
+    def get_current_year_start(self, user):
+        return self._current_year_start
+
     def _get_per_asset_usage(self, user):
+        users = [user]
+        anchor_date, period_start, subscription_interval = None, None, None
+        # Get the organization ID passed in from the query parameters
+        organization_id = self.context['request'].query_params.get('organization_id')
+        if organization_id:
+            try:
+                organization = Organization.objects.filter(
+                    owner__organization_user__user=user,
+                    id=organization_id,
+                )
+                # If the user is in an organization, get all org users so we can query their total org usage
+                organization_users = OrganizationUser.objects.filter(organization=organization).select_related('user')
+                users = [org_user.user.id for org_user in list(organization_users)]
+            except ObjectDoesNotExist:
+                # Couldn't find organization or organization users, raise an error
+                raise Http404
+            # Get the organization's subscription, if they have one
+            subscription = Subscription.objects.filter(
+                status__in=['active', 'past_due', 'trialing'],
```

Comment:
We should probably make this list a constant. Even though it's used just once, it's more readable to understand why these three strings are special. Just the variable name itself would be good documentation.